include/seastar/core/future.hh

```diff
@@ -1135,12 +1140,17 @@ public:
 
 private:
 
-    template <typename Func, typename Result = futurize_t<std::result_of_t<Func(future&&)>>>
+    template <typename Func, typename Result = futurize_t<std::result_of_t<Func(future)>>>
```

Comment:
I'm not sure why it gets switched back here.  Presumably, it's due to the removal of future(get_available_state()) below?

Reply:
It should be `future&&`. Looks I messed this up during rebase. Fixing.

Reply:
Fixed.